### ISSUE TITLE
Bullets ignore objects if they hit glass

### ISSUE THREAD

**Issue by Relyksboa** [CleverRaven/Cataclysm-DDA#37137]
Game version:
0.d-11393-g704f045
Operating system:
Windows
Tiles or curses:
Tiles
Mods active:
DDA
### Expected behavior
Vehicles to block bullets, even if the bullet is partially blocked by glass.
### Actual behavior
After passing through reinforced glass, the bullet phases through the vehicle.
### Steps to reproduce the behavior
1. spawn turret which can penetrate reinforced glass\
2. spawn a vehicle (note that damage does not occur to player due to the vehicle blocking the bullet)
3. Use the map editor to create a pane of reinforced glass in between the turret and the car
4. wait until it fires again 
5. Cry viciously as the bullet phases through your vehicle.

*screenshots of the process*
The initial Setup of the testing,  a m16a4 robot and a 4x4 car, newly created survivor
![initial set up](https://user-images.githubusercontent.com/52992503/72583557-99d23d00-389b-
11ea-8d93-aed6439ccdfe.PNG)
The damage from one shot of the previous robot, as blocked by the car.
![one shot of the robot](https://user-images.githubusercontent.com/52992503/72583563-9e96f100-389b-11ea-81d2-1123146849e6.PNG)
The introduction of reinforced glass to the experiment
![Set up with glass](https://user-images.githubusercontent.com/52992503/72583564-a191e180-389b-11ea-8e36-f864c5fea37e.PNG)
Phasing bullets as seen here, dealing 4 damage to the survivor, despite originally hitting the car.
![Miraculous bullet phasing technology](https://user-images.githubusercontent.com/52992503/72583568-a3f43b80-389b-11ea-9fb5-65f334f81c53.PNG)

**Comment by Night-Pryanik** [comment]
I don't think this is a bug at all. Bullet lost most of its energy when it hit reinforced glass, but it still has enough energy to make a hole in thin metal ordinary cars are made of.

**Comment by anothersimulacrum** [comment]
I agree this isn't an issue, unless the glass is not being damaged by the bullet.

**Comment by Relyksboa** [comment]
This is actually a bug as normally, the car would block the bullet. After it hits reinforced glass the bullet PHASES THROUGH the car and hits the target.

**Comment by Night-Pryanik** [comment]
Ok, tested and confirmed. 
Though I think "Normally the car would block projectiles" is not ok. And therefore bullet hitting you after reinforced glass is normal behavior (car's boards are too thin to provide enough protection), so we should investigate why bullets are fully blocked by car.

**Comment by Relyksboa** [comment]
@Night-Pryanik The issue is that the bullet is not colliding with the vehicle at all. It hits the glass and hits the player. But NOT the vehicle.

**Comment by Relyksboa** [comment]
@Night-Pryanik 
Here is a helpful diagram to explain the issue as cdda screenshots are not explaining enough about the issue.
![Helpful Diagram](https://user-images.githubusercontent.com/52992503/72687199-6e746b80-3ab0-11ea-81cf-e9358a23aa11.png)

**Comment by Night-Pryanik** [comment]
I got it, thank you.
